src/twigextensions/FetchTwigExtension.php

```diff
@@ -33,18 +33,23 @@ public function getFunctions()
       ];
   }
 
-  public function fetch($client, $method, $destination, $request = [], $parseJson = true)
+  public function fetch($client, $method, $destination, $request = [], $format = 'json')
   {
       $client = new \GuzzleHttp\Client($client);
 
       try {
 
         $response = $client->request($method, $destination, $request);
 
-        if ($parseJson) {
+        if ($format == 'json') {
             $body = json_decode($response->getBody(), true);
+        } elseif ($format == 'xml') {
+            $xmlbody = simplexml_load_string($response->getBody(), null, LIBXML_NOCDATA);
+
+            $json = json_encode($xmlbody);
```

Comment:
The $json variable is unnecessary here. Put `json_encode($xmlbody)` on the next line instead where `$json` is.

Reply:
Yeah I think that snippet is what Patrik is suggesting. A bit DRYer I guess, but I personally prefer it as you have it for readability... (If I'm misunderstanding your point @patrikalienus feel free to correct me)

Reply:
Yeh, it was more for readability. It was a bit rushed too at the time, so wasn't thinking to save a few lines. I guess where there isn't much in it, then @patrikalienus suggestion is still valid.